hack/convert-output-to-tsv.py

```diff
@@ -2,38 +2,163 @@
+                    # remove 'hles_' from the table_name
+                    pk_name = table_name[5:] + '_id'
+                    entity_name = pk_prefix + pk_name
+                    row[entity_name] = row.pop(pk_name)
+                    # store data
+                    column_set.update(row.keys())
+                    row_list.append(row)
+                # hles_cancer_condition: read dog_id, copy and write out as pk_name
+                elif (table_name == "hles_cancer_condition"):
+                    entity_name = pk_prefix + table_name + '_id'
+                    # copy the dog_id
+                    row[entity_name] = row.get('dog_id')
+                    # store data
+                    column_set.update(row.keys())
+                    row_list.append(row)
+                # hles_health_condition: read + copy dog_id and hs_condition and hs_condition_is_congenital
+                # concatenate and write out as pk_name
+                elif (table_name == "hles_health_condition"):
+                    entity_name = pk_prefix + table_name + '_id'
+                    # grab the congenital flag and convert to int
+                    congenital_flag = row.get('hs_condition_is_congenital')
+                    if congenital_flag == True:
+                        congenital_flag = 1
+                    elif congenital_flag == False:
+                        congenital_flag = 0
+                    else:
+                        print ("Error, 'hs_condition_is_congenital' is not populated in {0}"
+                            .format(table_name))
+                    row[entity_name] = ('%s-%s-%s' % 
+                        (row.get('dog_id'), row.get('hs_condition'), congenital_flag))
+                    # store data
+                    column_set.update(row.keys())
+                    row_list.append(row)
+                else:
+                    print("Unrecognized table: %s" % table_name)
+
+    # make sure pk is the first column
+    # pop out the PK, will be splitting this set out later
+    column_set.remove(entity_name)
+    sorted_column_set = sorted(list(column_set))    
+    sorted_column_set.insert(0, entity_name)
 
+    # provide some stats
+    col_count = len(sorted_column_set)
+    print("...%s contains %s rows and %s columns" % 
+        (table_name, row_count, col_count))
     # output to tsv
-    output_location = output_dir + '/' + output_name + '.tsv'
-    with open(output_location, 'w') as output_file:
-        dw = csv.DictWriter(output_file, sorted(list(column_set)), delimiter='\t')
-        dw.writeheader()
-        dw.writerows(row_list)
+    # 512 column max limit per request (upload to workspace) 
+    if (col_count > 512):
+        # calculate chunks needed - each table requires the PK
+        total_col_count = ceil(col_count/512)+col_count-1
+        chunks = ceil(total_col_count/512)
+        print("...Splitting %s into %s files" % (table_name, chunks))
```

Comment:
can use the print(f"...Splitting {table_name} into {chunks} files") syntax, could be a little shorter but this is fine too. Your call.